src/test/java/net/openhft/chronicle/bytes/perf/ContentEqualJLBH.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2016-2022 chronicle.software
+ *
+ *     https://chronicle.software
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.openhft.chronicle.bytes.perf;
+
+import net.openhft.chronicle.bytes.Bytes;
+import net.openhft.chronicle.bytes.BytesUtil;
+import net.openhft.chronicle.bytes.internal.BytesInternal;
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.jlbh.JLBH;
+import net.openhft.chronicle.jlbh.JLBHOptions;
+import net.openhft.chronicle.jlbh.JLBHTask;
+import net.openhft.chronicle.jlbh.TeamCityHelper;
+
+import java.util.function.Supplier;
+
+public class ContentEqualJLBH implements JLBHTask {
+    private static final int ITERATIONS = 5_000_000;
+    private final Bytes<?> bytes1;
+    private final Bytes<?> bytes2;
+    private JLBH jlbh;
+
+    public ContentEqualJLBH(Supplier<Bytes<?>> bytesSupplier) {
+        bytes1 = bytesSupplier.get();
+        bytes2 = bytesSupplier.get();
+        BytesUtil.reverse(bytes2, 0);
+    }
+
+    @Override
+    public void init(JLBH jlbh) {
+        this.jlbh = jlbh;
+    }
+
+    @Override
+    public void complete() {
+        // this is horrible - it would be easier if Bootstrap.getMajorVersion0 was exposed
+        String javaVersion = Jvm.isJava9Plus() ? Jvm.isJava15Plus() ? "17" : "11" : "8";
+        TeamCityHelper.teamCityStatsLastRun(getClass().getSimpleName() + "-Java" + javaVersion + "-" + bytes1.length(), jlbh, ITERATIONS, System.out);
+    }
+
+    @Override
+    public void run(long startTimeNS) {
+        boolean matching = BytesInternal.contentEqual(bytes1, bytes2);
+        if (matching)
+            throw new IllegalStateException();
+        jlbh.sampleNanos(System.nanoTime() - startTimeNS);
```

Comment:
Would it be better to have a few calls to contentEquals that varied in equality (e.g. below) because sometimes the performance of the comparisons depends on how much of the string is equal?

|lhs              | rhs         |
|---------------|--------------|
|`xxxxxxx`   | `xxxxxxx`
|`xxxxxxx`   | `xxxxxxo`
|`xxxxxxx`   | `xxxxxoo`
|`xxxxxxx`   | `xxxxooo`
|`xxxxxxx`   | `xxxoooo`
|`xxxxxxx`   | `xxooooo`
|`xxxxxxx`   | `xoooooo`
|`xxxxxxx`   | `ooooooo`

Reply:
Great idea - done